### PR TITLE
Fix tests where no network is available

### PR DESCRIPTION
**Pull request by etbuira**
This follows PR #1056

### CODE REVIEW

**Comment by Lukasa** [comment]
Ok, at this point I have to ask you to stop closing PRs and opening new ones. If you simply push new commits, or rebase your current ones, on the same branch the pull request will automatically update. It becomes much easier to track work if you do it that way around. =)

File: test/__init__.py [hunk 1]
```diff
@@ -100,18 +100,18 @@ def wrapper(*args, **kwargs):
...
             # connection to the TARPIT_HOST. This fails if you are in a place
             # without an Internet connection, so we skip the test in that case.
             if _is_unreachable_err(e):
                 raise SkipTest(msg)
             raise
-        except MaxRetryError as e:
-            if _is_unreachable_err(e.reason):
-                raise SkipTest(msg)
-            raise
+        return test(*args, **kwargs)
```
Comment (Lukasa) [inline test/__init__.py]: So we haven't answered my question on the previous PR: *why* is the current decorator insufficient?

Comment (etbuira) [inline test/__init__.py]: This is explained in the commit message

Comment (Lukasa) [inline test/__init__.py]: Ok, so in that case I'd like to change the logic here. I'd like to do this check only once and remember the result. =) Save ourselves from doing this too often.

**Pull request closed without merging**